File: app/java/app/src/main/java/com/example/accounting/financial/BalanceSheet.java
package com.example.accounting.financial;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.util.List;

/**
 * 貸借対照表
 */
public class BalanceSheet {
    private LocalDate asOfDate;  // 基準日
    private List<BalanceSheetItem> assets;  // 資産
    private List<BalanceSheetItem> liabilities;  // 負債
    private List<BalanceSheetItem> equity;  // 純資産
    private BigDecimal totalAssets;  // 資産合計
    private BigDecimal totalLiabilities;  // 負債合計
    private BigDecimal totalEquity;  // 純資産合計
    private BigDecimal totalLiabilitiesAndEquity;  // 負債・純資産合計

    @SuppressWarnings("checkstyle:ParameterNumber")
    public BalanceSheet(LocalDate asOfDate, List<BalanceSheetItem> assets,
                        List<BalanceSheetItem> liabilities, List<BalanceSheetItem> equity,
                        BigDecimal totalAssets, BigDecimal totalLiabilities,
                        BigDecimal totalEquity, BigDecimal totalLiabilitiesAndEquity) {
        this.asOfDate = asOfDate;
        this.assets = assets;
        this.liabilities = liabilities;
        this.equity = equity;
        this.totalAssets = totalAssets;
        this.totalLiabilities = totalLiabilities;
        this.totalEquity = totalEquity;
        this.totalLiabilitiesAndEquity = totalLiabilitiesAndEquity;
    }

    public LocalDate getAsOfDate() {
        return asOfDate;
    }

    public void setAsOfDate(LocalDate asOfDate) {
        this.asOfDate = asOfDate;
    }

    public List<BalanceSheetItem> getAssets() {
        return assets;
    }

    public void setAssets(List<BalanceSheetItem> assets) {
        this.assets = assets;
    }

    public List<BalanceSheetItem> getLiabilities() {
        return liabilities;
    }

    public void setLiabilities(List<BalanceSheetItem> liabilities) {
        this.liabilities = liabilities;
    }

    public List<BalanceSheetItem> getEquity() {
        return equity;
    }

    public void setEquity(List<BalanceSheetItem> equity) {
        this.equity = equity;
    }

    public BigDecimal getTotalAssets() {
        return totalAssets;
    }

    public void setTotalAssets(BigDecimal totalAssets) {
        this.totalAssets = totalAssets;
    }

    public BigDecimal getTotalLiabilities() {
        return totalLiabilities;
    }

    public void setTotalLiabilities(BigDecimal totalLiabilities) {
        this.totalLiabilities = totalLiabilities;
    }

    public BigDecimal getTotalEquity() {
        return totalEquity;
    }

    public void setTotalEquity(BigDecimal totalEquity) {
        this.totalEquity = totalEquity;
    }

    public BigDecimal getTotalLiabilitiesAndEquity() {
        return totalLiabilitiesAndEquity;
    }

    public void setTotalLiabilitiesAndEquity(BigDecimal totalLiabilitiesAndEquity) {
        this.totalLiabilitiesAndEquity = totalLiabilitiesAndEquity;
    }
}

File: app/rust/accounting-system/src/domain/financial/balance_sheet.rs
use chrono::NaiveDate;
use rust_decimal::Decimal;
use serde::{Deserialize, Serialize};

/// 貸借対照表の項目
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct BalanceSheetItem {
    /// 勘定科目コード
    pub account_code: String,

    /// 勘定科目名
    pub account_name: String,

    /// 残高
    pub balance: Decimal,

    /// 構成比率（%）
    pub percentage: Decimal,
}

/// 貸借対照表
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct BalanceSheet {
    /// 基準日
    pub as_of_date: NaiveDate,

    /// 資産項目
    pub assets: Vec<BalanceSheetItem>,

    /// 負債項目
    pub liabilities: Vec<BalanceSheetItem>,

    /// 純資産項目
    pub equity: Vec<BalanceSheetItem>,

    /// 資産合計
    pub total_assets: Decimal,

    /// 負債合計
    pub total_liabilities: Decimal,

    /// 純資産合計
    pub total_equity: Decimal,

    /// 負債・純資産合計
    pub total_liabilities_and_equity: Decimal,
}
